plugins/outputs/dynatrace/dynatrace.go

```diff
@@ -101,163 +90,97 @@ func (d *Dynatrace) Description() string {
 	return "Send telegraf metrics to a Dynatrace environment"
 }
 
-// Normalizes a metric keys or metric dimension identifiers
-// according to Dynatrace format.
-func (d *Dynatrace) normalize(s string, max int) (string, error) {
-	s = reNameAllowedCharList.ReplaceAllString(s, "_")
-
-	// Strip Digits and underscores if they are at the beginning of the string
-	normalizedString := strings.TrimLeft(s, "_0123456789")
-
-	for strings.HasPrefix(normalizedString, "_") {
-		normalizedString = normalizedString[1:]
-	}
-
-	if len(normalizedString) > max {
-		normalizedString = normalizedString[:max]
-	}
-
-	for strings.HasSuffix(normalizedString, "_") {
-		normalizedString = normalizedString[:len(normalizedString)-1]
-	}
-
-	normalizedString = strings.ReplaceAll(normalizedString, "..", "_")
-
-	if len(normalizedString) == 0 {
-		return "", fmt.Errorf("error normalizing the string: %s", s)
-	}
-	return normalizedString, nil
-}
```

Comment:
The normalization is basically the same except the utility library handles more odd corner cases that we've discovered when implementing other exporters like the opentelemetry SDK and collector exporters. We've taken what we learned writing all these exporters and made the utility library so it will be handled more consistently for our users.

Reply:
No the behavior is unchanged. The serialization logic is just moved into a shared library so we can keep behavior consistent between this and other exporters like opentelemetry.

Reply:
Ok that makes sense